src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -754,6 +754,135 @@ public void MultithreadedAnalyzeCommandBase_EndToEndMultithreadedAnalysis()
+                    fileSize = randomFileSize,
+                    maxFileSize = int.MaxValue
+                },
+                new {
+                    expectedExitReason = ExitReason.InvalidCommandLineOption,
+                    fileSize = long.MaxValue,
+                    maxFileSize = int.MinValue
+                },
+                new {
+                    expectedExitReason = ExitReason.InvalidCommandLineOption,
+                    fileSize = long.MaxValue,
+                    maxFileSize = 0
+                },
+                new {
+                    expectedExitReason = ExitReason.NoValidAnalysisTargets,
+                    fileSize = long.MaxValue,
+                    maxFileSize = int.MaxValue
+                },
+            };
+
+            foreach (dynamic testCase in testCases)
+            {
+                string specifier = "*.xyz";
+
+                int filesCount = 10;
+                var files = new List<string>();
+                for (int i = 0; i < filesCount; i++)
+                {
+                    files.Add(Path.GetFullPath($@".{Path.DirectorySeparatorChar}File{i}.txt"));
+                }
+
+                var propertiesDictionary = new PropertiesDictionary();
+                propertiesDictionary.SetProperty(TestRule.ErrorsCount, (uint)15);
+                propertiesDictionary.SetProperty(TestRule.Behaviors, TestRuleBehaviors.LogError);
+
+                using var tempFile = new TempFile(".xml");
+                propertiesDictionary.SaveToXml(tempFile.Name);
+
+                var mockStream = new Mock<Stream>();
+                mockStream.Setup(m => m.CanRead).Returns(true);
+                mockStream.Setup(m => m.CanSeek).Returns(true);
+                mockStream.Setup(m => m.ReadByte()).Returns('a');
+
+                var mockFileSystem = new Mock<IFileSystem>();
+                mockFileSystem.Setup(x => x.DirectoryExists(It.IsAny<string>())).Returns(true);
+                mockFileSystem.Setup(x => x.DirectoryGetFiles(It.IsAny<string>(), specifier)).Returns(files);
+                mockFileSystem.Setup(x => x.FileExists(It.Is<string>(s => s.EndsWith(specifier)))).Returns(true);
+                mockFileSystem.Setup(x => x.DirectoryEnumerateFiles(It.IsAny<string>(),
+                                                                    It.IsAny<string>(),
+                                                                    It.IsAny<SearchOption>())).Returns(files);
+                mockFileSystem.Setup(x => x.FileOpenRead(It.IsAny<string>())).Returns(mockStream.Object);
+                mockFileSystem.Setup(x => x.FileExists(tempFile.Name)).Returns(true);
+                mockFileSystem.Setup(x => x.GetFileSize(It.IsAny<string>())).Returns(testCase.fileSize);
+
+                Output.WriteLine($"The seed that will be used is: {TestRule.s_seed}");
+
+                var options = new TestAnalyzeOptions
+                {
+                    Threads = 10,
```

Comment:
>Threads = 10 [](http://example.com/codeflow?start=20&length=12)

Why are you opting into this behavior, and enabling hashes, etc? This is a nice test, but it's doing a lot, why is that? Can't we make this a bit more focused on the matter at hand, file size validation?

When you start adding things, you increase performance costs, but you also risk introducing non-obvious breaks in tests. So, let's say we had a problem introduced by our hashing argment to --insert. This test may fail, and we'd all be confused why the file size test broken.

This is just an example. But the principle here is that we want targeted unit tests that directly focus on some test work, as far as possible

It isn't always possible to create a clean union. But for this example, I definitely don't see why we'd explicitly set any args, would prefer defaults and minimal options otherwise.

Reply:
Hashes, etc were part of the original customer escalation. It's fairly conclusive that with and without these the repro and fix are the same. I've removed.